Description: compatibility with OpenSSL 3
 OpenSSL 3 no longer supports an ERR_GET_FUNC() macro, so remove the reference
 from the code.
Author: Steve Langasek <steve.langasek@ubuntu.com>
Last-Update: 2021-12-08
Forwarded: no

Index: trustedqsl-2.5.9/src/openssl_cert.cpp
===================================================================
--- trustedqsl-2.5.9.orig/src/openssl_cert.cpp
+++ trustedqsl-2.5.9/src/openssl_cert.cpp
@@ -4285,7 +4285,7 @@
 tqsl_ssl_error_is_nofile() {
 	unsigned long l = ERR_peek_error();
 	if (tQSL_Error == TQSL_OPENSSL_ERROR &&
-		ERR_GET_LIB(l) == ERR_LIB_SYS && ERR_GET_FUNC(l) == SYS_F_FOPEN)
+		ERR_GET_LIB(l) == ERR_LIB_SYS)
 		return 1;
 	if (tQSL_Error == TQSL_SYSTEM_ERROR && tQSL_Errno == ENOENT)
 		return 1;
